lib/stages/visTranspose.cpp

```diff
@@ -223,9 +272,17 @@ void visTranspose::main_thread() {
         auto frame = visFrameView(in_buf, frame_id);
 
         if (frame.dataset_id != ds_id) {
-            FATAL_ERROR("Dataset ID of incoming frames changed from {} to {}. Changing  ID "
-                        "not supported, exiting...",
-                        ds_id, frame.dataset_id);
+            if (frame.fpga_seq_length == 0) {
+                INFO("Got an empty frame.");
+            } else {
+                // TODO assuming that dataset ID changes here never change dataset dimensions
+                INFO("Dataset ID has changed from {} to {}. Getting base dataset ID from "
+                     "broker...",
+                     ds_id, frame.dataset_id);
+                ds_id = frame.dataset_id;
+                base_ds_id = base_dset(ds_id);
+                INFO("Got base dataset ID {}.", base_ds_id);
```

Comment:
This seems like it should be `DEBUG`, but I leave this decision to you, since you probably have a better idea of what is useful.